packages/crypto/src/crypto.ts

```diff
@@ -466,3 +568,251 @@ const bigIntToHex = (num: bigint, length: number): string => {
+      "expected signer key does not match signer key from bundle"
+    );
+  }
+
+  const encryptionQuorumPublicBuf = new Uint8Array(
+    uint8ArrayFromHexString(enclaveQuorumPublic)
+  );
+  const quorumKey = await loadQuorumKey(encryptionQuorumPublicBuf);
+  if (!quorumKey) {
+    throw new Error("failed to load quorum key");
+  }
+
+  // The ECDSA signature is ASN.1 DER encoded but WebCrypto uses raw format
+  const publicSignatureBuf = fromDerSignature(publicSignature);
+  const signedDataBuf = uint8ArrayFromHexString(signedData);
+  return await crypto.subtle.verify(
+    { name: "ECDSA", hash: { name: "SHA-256" } },
+    quorumKey,
+    publicSignatureBuf,
+    signedDataBuf
+  );
+};
+
+/**
+ * Converts an ASN.1 DER-encoded ECDSA signature to the raw format that WebCrypto uses.
+ *
+ * @param {string} derSignature - The DER-encoded signature.
+ * @returns {Uint8Array} - The raw signature.
+ */
+const fromDerSignature = (derSignature: string) => {
+  const derSignatureBuf = uint8ArrayFromHexString(derSignature);
+
+  // Check and skip the sequence tag (0x30)
+  let index = 2;
+
+  // Parse 'r' and check for integer tag (0x02)
+  if (derSignatureBuf[index] !== 0x02) {
+    throw new Error(
+      "failed to convert DER-encoded signature: invalid tag for r"
+    );
+  }
+  index++; // Move past the INTEGER tag
+  const rLength = derSignatureBuf[index];
+  index++; // Move past the length byte
+  const r = derSignatureBuf.slice(index, index + rLength!);
+  index += rLength!; // Move to the start of s
+
+  // Parse 's' and check for integer tag (0x02)
+  if (derSignatureBuf[index] !== 0x02) {
+    throw new Error(
+      "failed to convert DER-encoded signature: invalid tag for s"
+    );
+  }
+  index++; // Move past the INTEGER tag
+  const sLength = derSignatureBuf[index];
+  index++; // Move past the length byte
+  const s = derSignatureBuf.slice(index, index + sLength!);
+
+  // Normalize 'r' and 's' to 32 bytes each
+  const rPadded = normalizePadding(r, 32);
+  const sPadded = normalizePadding(s, 32);
+
+  // Concatenate and return the raw signature
+  return new Uint8Array([...rPadded, ...sPadded]);
+};
+
+/**
+ * Loads an ECDSA public key from a raw format for signature verification.
+ *
+ * @param {Uint8Array} quorumPublic - The raw public key bytes.
+ * @returns {Promise<CryptoKey>} - The imported ECDSA public key.
+ */
+const loadQuorumKey = async (quorumPublic: Uint8Array): Promise<CryptoKey> => {
+  return await crypto.subtle.importKey(
+    "raw",
+    quorumPublic,
+    {
+      name: "ECDSA",
+      namedCurve: "P-256",
+    },
+    true,
+    ["verify"]
+  );
+};
+
+/**
+ * Decodes a private key based on the specified format.
+ *
+ * @param {string} privateKey - The private key to decode.
+ * @param {string} keyFormat - The format of the private key (e.g., "SOLANA", "HEXADECIMAL").
+ * @returns {Uint8Array} - The decoded private key.
+ */
+
+export const decodeKey = (privateKey: string, keyFormat: any): Uint8Array => {
```

Comment:
should we export these functions if they're not gonna be used externally? I think it's safer to export the minimal surface first. If we get feedback that these internal functions would be helpful we can certainly export them. But I'd rather err on the side of being conservative with what we export / promise as far as package interface goes.

Reply:
Agreed!